test/verify/check-login

```diff
@@ -561,5 +561,135 @@ account    required     pam_succeed_if.so user ingroup %s""" % m.get_admin_group
+                      "s/sufficient/[success=1 default=ignore]/\n"
+                      "aauth [default=die] pam_faillock.so authfail audit deny=4 unlock_time=600\n"
+                      "aauth sufficient pam_faillock.so authsucc audit deny=4 unlock_time=600\n"
+                      "}' /etc/pam.d/password-auth")
+
+        m.execute("grep -r '{}' /etc/pam.d".format(module))
+
+        m.start_cockpit()
+
+        # pam_faillock tries to chown the file to be owned by the user itself,
+        # which is currently an selinux fail on the following distros:
+        if m.image in ['centos-8-stream', 'fedora-31', 'fedora-32', 'rhel-8-2', 'rhel-8-2-distropkg', 'rhel-8-3']:
+            # https://bugzilla.redhat.com/show_bug.cgi?id=1836182
+            self.allow_journal_messages('.*type=1400.*avc:  denied  { chown }.*comm="cockpit-session".*')
+
+        if module == 'pam_tally2':
+            # at first, we should have no log file
+            m.execute("! test -f {}".format(logfile('admin')))
+
+        # make sure we can still login
+        b.login_and_go("/system")
+        b.logout()
+
+        if module == 'pam_tally2':
+            # now we should have it
+            m.execute("test -f {}".format(logfile('admin')))
+
+        # and it should show zero fails
+        expect_fail_count(0, must_exist=False)
+
+        # try three bogus login attempts
+        for n in [1, 2, 3]:
+            expect_failed_login("admin", "bad", n)
+
+        # make sure we can still login
+        expect_successful_login("admin", "foobar")
+
+        # after the success, try three more bogus login attempts
+        for n in [1, 2, 3]:
+            expect_failed_login("admin", "bad", n)
+        expect_successful_login("admin", "foobar")
+
+        # now try four, which should lock the account
+        for n in [1, 2, 3, 4]:
+            expect_failed_login("admin", "bad", n)
+
+        # logging in should fail now
+        if module == 'pam_tally2': # this is counted differently by each module
+            n += 1
+        expect_failed_login("admin", "foobar", n)
+
+        # having given the correct password the last time should not have helped things
+        if module == 'pam_tally2':
+            n += 1
+        expect_failed_login("admin", "foobar", n)
+
+        # but we can reset the lockout
+        m.execute(module + " --reset --user admin")
+        expect_fail_count(0)
+
+        # and login again
+        expect_successful_login("admin", "foobar")
+
+        # fedora-coreos logs in via sshd, which normally forbids root logging
+        # in with a password, so change that before we proceed
+        if m.image in ['fedora-coreos']:
+            self.restore_file("/etc/ssh/sshd_config", post_restore_action="systemctl restart sshd.service")
+            m.execute("sed -i 's/PermitRootLogin prohibit-password/PermitRootLogin yes/' /etc/ssh/sshd_config")
+            m.execute("systemctl restart sshd.service")
```

Comment:
Simplify this with

    m.sed_file('s/PermitRootLogin prohibit-password/PermitRootLogin yes/', '/etc/ssh/sshd_config', 'systemctl try-restart sshd.service')

Reply:
I was wondering if I should make a convenience wrapper here.  I even grepped for one, briefly.  I guess it already exists.  Thanks!